test/test-communities.rb

```diff
@@ -0,0 +1,290 @@
+class CommunitiesTest < Test::Unit::TestCase
+  def setup
+    @dataset = Datasets::Communities.new
+  end
+
+  def record(*args)
+    Datasets::Communities::Record.new(*args)
+  end
+
+  test('#each') do
+    records = @dataset.each.to_a
+    assert_equal([
+                    1994,
+                    {
+                      :state => "8",
+                      :county => "?",
+                      :community => "?",
+                      :communityname => "Lakewoodcity",
+                      :fold => "1",
+                      :population => "0.19",
```

Comment:
Can we convert floating numbers to `Float` instead of `String`?

Reply:
Thanks for the review. 

Converted floating numbers to numeric values if possible based on the data set description provided at http://archive.ics.uci.edu/ml/machine-learning-databases/communities/communities.names . except `community` and `community` both of them are defined as numeric but actually they have "?" string character.

Reply:
OK.
`"?"` means that it's a missing value. So we can convert `"?"` as `nil` in Ruby.